File: FrontEnd/src/models/onLeaveDTO.ts
import { WorkflowInstancesDTO } from "@models/workflowInstancesDTO";
import { WorkflowStepDTO } from "@models/workflowStepDTO";

export interface OnLeaveDTO {
  id: string;
  createdDate: Date;
  status: number | null;
  description: string | null;
  idUserRequest: string;
  idEmployee: string;
  idUnit: string;
  unitName: string | null;
  workflowInstances: WorkflowInstancesDTO[];
  countWorkFlowStep: number;
  currentWorkFlowStep: WorkflowStepDTO;
  attachments: string | null;
  fromDate: Date;
  toDate: Date;
  unPaidLeave: boolean;
}

File: BackEnd/ExamManagement/DPD.HR.Business.Dto/Dto/OnLeaveDto.cs
using DPD.HumanResources.Entities.Entities;

namespace DPD.HumanResources.Dtos.Dto;

public class OnLeaveDto
{
    public Guid Id { get; set; }
    public DateTime? CreatedDate { get; set; }
    public int? Status { get; set; }
    public string? Description { get; set; }
    public Guid IdUserRequest { get; set; }
    public Guid IdEmployee { get; set; }
    public Guid IdUnit { get; set; }
    public string? UnitName { get; set; }
    public string? Attachments {get;set;}
    public DateTime? FromDate {get;set;}
    public DateTime? ToDate {get;set;}
    public bool UnPaidLeave {get;set;}


    public List<WorkflowInstances>? WorkflowInstances { get; set; }
    public int CountWorkFlowStep { get; set; }
    public WorkflowStep? CurrentWorkFlowStep { get; set; }
    public Guid? IdFile { get; set; }

}